packages/cxl-ui/src/components/cxl-details/cxl-details-checkout.js

```diff
@@ -0,0 +1,64 @@
+import '@conversionxl/cxl-lumo-styles';
+import { registerGlobalStyles } from '@conversionxl/cxl-lumo-styles/src/utils';
+import { Details } from '@vaadin/details/src/vaadin-details';
+import { customElement } from 'lit/decorators.js';
+import cxlDetailsCheckoutGlobalStyles from '../../styles/global/cxl-details-checkout-css.js';
+
+@customElement('cxl-details-checkout')
+export class CXLDetailsCheckoutElement extends Details {
+  O;
+  ready() {
+    super.ready();
+
+    registerGlobalStyles(cxlDetailsCheckoutGlobalStyles, {
+      moduleId: 'cxl-details-checkout-global',
+    });
+
+    this.addEventListener('opened-changed', (e) => {
+      // Fire only when closing.
+      if (!e.detail.value) {
+        this.__onOpenedChanged(e);
+      }
+    });
+  }
+
+  __onOpenedChanged(e) {
+    switch (this.getAttribute('step')) {
+      case 'billing-details': {
+        // Gather fields.
+        const first = this.querySelector('#billing-first-name').value;
+        const last = this.querySelector('#billing-last-name').value;
```

Comment:
Fairly strong coupling. I still wonder if we can we pass field list in declaratively like `<cxl-details-checkout data-summary-fields="billing-first-name billing-last-name">` etc.

I see simply space-separated convention shown at https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*#usage

(We probably don't need anything more complex like JSON structures in a data attribute.. right?)